Add CreatorPro / Makerbot / QIDI / etc and Mightboard RevE support

mikeshub opened the pull request:
This commit adds a few features and fixes a bug. The bug was in the MAXxxxx thermocouple temp reporting. Support for 3 wire HD44780 was added. Support for PCA9533 RGB driver was added. Configuration example file was created.

### Requirements

You'll need a couple of libraries I extracted from the Sailfish firmware. I have hosted them on my github:

https://github.com/mikeshub/SailfishRGB_LED

https://github.com/mikeshub/SailfishLCD

### Description

I got CreatorPro and I didn't like Sailfish. Right when I got it I spent a bit of time looking at porting Marlin and wasn't up to the task. Then last week I saw the great work started by Mateusz Krawczuk and was lucky enough to download what he started off his website (I think I downed it by doing that sorry Mateusz ). Over the weekend I decided to go ahead and bring support for CreatorPro / Makerbot / QIDI / etc etc etc the to completion. 

### Benefits

This let's others who are also massively frustrated by Sailfish not have to deal with it anymore on Mighty Board based printers. Dual extrusion worked right on Simplify and the CreatorPro, but it's dual nozzle support is way behind Cura. The good folks Ultimaker rightfully have no interest in adding support to abandonware like Sailfish.

### Related Issues

Bug fixed:

There was an issue with the MAXxxxx thermocouple temp sensing where the temps for the nozzles would get mixed up. 

Features added:

3 wire HD44780 support

PCA9533 RGB LED driver support

mikeshub committed:
Marlin 2.0 on Creator Pro

This commit adds a few features and fixes a bug. The bug was in the MAXxxxx thermocouple temp reporting. Support for 3 wire HD44780 was added. Support for PCA9533 RGB driver was added. Configuration example file was created.

mikeshub committed:
Update default configuration.h

mikeshub commented:
Checks failed, probably because my new libraries. I did not do this coding in platformIO.

Ludy87 commented:
add in `platformIO.ini`

```ini
lib_deps =
  https://github.com/mikeshub/SailfishLCD.git
  https://github.com/mikeshub/SailfishRGB_LED.git
```

and try again

mikeshub committed:
Add libraries to lib_debs

mikeshub commented:
Hopefully this works. If not I am pretty sure it will be because I have used AVR specific code in the LED driver. I'll switch it over to Wire.h tonight if that is the case.

mikeshub commented:
OK so I have two ideas about why that failed. One is am I going to need to add some json file to the libraries for them to work?

The second would be that I have used AVR specific code in the LED driver so I will switch the LED driver over to wire.h.

Ludy87 commented:
You overwrite the default config. It's bad idea enable `FF_INTERFACEBOARD` and `PCA9533`

mikeshub commented:
Good catch. I went ahead and switched the LED driver over to Wire.h as well. Fingers crossed.

mikeshub committed:
Comment out migthyboard related defines from default config file

mikeshub commented:
OK easy fix in the Max code. Next one should get it.

mikeshub committed:
Fix Temperature::read_max6675 with single IC

mikeshub committed:
Commit to start autobuild process Travis CI build using updated libs

The LCD and LED drivers were using _delay_microseconds whereas they should have been using delayMicroseconds

mikeshub commented:
Thanks again for the tips Ludy87.

Ludy87 commented:
No problem 👍

mikeshub committed:
Add SlowSoftI2CRepo

Add SlowSoftI2C repo to platformio.ini

mikeshub committed:
Fix for filament change preheat temp and center at 0,0 homing

mikeshub committed:
I accidentally overwrote the default config on the last commit

mikeshub commented:
I was a little to fast with that one commit. Should have double checked to make sure my config headers weren't checked. The last one restored the default config.

mikeshub committed:
Update config.h to use more reasonable velocities

mikeshub commented:
OK I've gotten a few good prints out of the machine today. 5143bbc is the last commit on the PR.

thinkyhead committed:
Update some things

mikeshub commented:
Looks good. My only suggestion would be to remove line 193 from pins_MIGHTYBOARD_REVE.h entirely as the RGB LEDs are I2C and not on a PWM brightness control.

thinkyhead committed:
Update temperature.cpp

thinkyhead committed:
Update pins_MIGHTYBOARD_REVE.h

thinkyhead commented:
> My only suggestion would be to remove line 193 from `pins_MIGHTYBOARD_REVE.h` entirely as the RGB LEDs are I2C and not on a PWM brightness control.

The case light pin definition (and others) exist for those users who want to implement those features and need a suggested default pin. If RGB LEDs or Neopixels are used, those features are associated with RGBW or i2c pins. And the case light feature, when enabled, is associated with a single PWM pin. It is possible to have all these features enabled at once, so long as there are no pin conflicts set up between them.

mikeshub commented:
The case light pin will cause conflicts. Pin 44 is the fan pin labeled extra fet on the schematic.

chepo92 commented:
Great work!, I'll give it a try, 
I had an issue compiling, in Arduino 1.8.9
In file digipot_mcp4018.cpp  Include "utility/twi.h" File doesn't exist 
had to comment that line 29. Include "utility/twi.h"

chepo92 commented:
Extruder 0 wasn't heating, solved in #15607,  it's an arduino core related issue,
now to start heating you have to M42 P16 S255 to enable a power safety latch and optocouples
